src/backend/Shared/EventBus/EventBus.ServiceBus/EventBusServiceBus.cs

```diff
@@ -0,0 +1,199 @@
+			byte[] body = Encoding.UTF8.GetBytes(jsonMessage);
+
+			Message message = new Message
+			{
+				MessageId = Guid.NewGuid().ToString(),
+				Body = body,
+				Label = eventName,
+			};
+
+			_persisterConnection
+				.CreateModel()
+				.SendAsync(message)
+				.GetAwaiter()
+				.GetResult();
+		}
+		#endregion
+
+		#region Subscribe
+		public void SubscribeDynamic<TH>(string eventName)
+			where TH : IDynamicIntegrationEventHandler
+		{
+			_subsManager.AddDynamicSubscription<TH>(eventName);
+		}
+
+		public void Subscribe<T, TH>()
+			where T : IntegrationEvent
+			where TH : IIntegrationEventHandler<T>
+		{
+			string eventName = typeof(T).Name.Replace(INTEGRATION_EVENT_SUFFIX, "");
+
+			bool containsKey = _subsManager.HasSubscriptionsForEvent<T>();
+			if (!containsKey)
+			{
+				_subscriptionClient.AddRuleAsync(new RuleDescription
+				{
+					Filter = new CorrelationFilter { Label = eventName },
+					Name = eventName
+				}).GetAwaiter().GetResult();
+			}
+
+			_subsManager.AddSubscription<T, TH>();
+		}
+		#endregion
+
+		#region Unsubscribe
+		public void UnsubscribeDynamic<TH>(string eventName)
+			where TH : IDynamicIntegrationEventHandler
+		{
+			_subsManager.RemoveDynamicSubscription<TH>(eventName);
+		}
+
+		public void Unsubscribe<T, TH>()
+			where T : IntegrationEvent
+			where TH : IIntegrationEventHandler<T>
+		{
+			string eventName = typeof(T).Name.Replace(INTEGRATION_EVENT_SUFFIX, "");
+
+			_subscriptionClient
+				.RemoveRuleAsync(eventName)
+				.GetAwaiter()
+				.GetResult();
+
+			_subsManager.RemoveSubscription<T, TH>();
+		}
+		#endregion
+
+		#region Consume Message
+		private void AddIntegrationEventHandler()
+		{
+			_subscriptionClient.RegisterMessageHandler(
+				async (message, token) =>
+				{
+					string eventName = $"{message.Label}{INTEGRATION_EVENT_SUFFIX}";
+					string messageData = Encoding.UTF8.GetString(message.Body);
+
+					// Complete the message so that it is not received again.
+					if (await ProcessEvent(eventName, messageData))
+					{
+						await _subscriptionClient.CompleteAsync(message.SystemProperties.LockToken);
+					}
+				},
+				new MessageHandlerOptions((e) => Task.CompletedTask) { MaxConcurrentCalls = 10, AutoComplete = false });
+		}
+
+		private async Task<bool> ProcessEvent(string eventName, string message)
+		{
+			if (!_subsManager.HasSubscriptionsForEvent(eventName)) return false;
+
+			using (ILifetimeScope scope = _autofac.BeginLifetimeScope(AUTOFAC_SCOPE_NAME))
+			{
+				IEnumerable<SubscriptionInfo> subscriptions = _subsManager.GetHandlersForEvent(eventName);
+
+				foreach (SubscriptionInfo subscription in subscriptions)
+				{
+					object handler = scope.ResolveOptional(subscription.HandlerType);
+					if (handler == null) continue;
+
+					if (subscription.IsDynamic)
+					{
+						await ProcessDynamicSubscription(handler, message);
+					}
+					else
+					{
+						await ProcessStaticSubscription(handler, eventName, message);
```

Comment:
![Codacy](https://app.codacy.com/assets/images/favicon.png) Issue found: [Add '.ConfigureAwait(false)' to this call to allow execution to continue in any thread.](https://app.codacy.com/manual/iamprovidence/Lama/pullRequest?prid=5244170)